[#13]add ul component

yg1110 opened the pull request:
- 채용 프로그램 ul 컴포넌트 추가.
- our-feature ul 컴포넌트 추가.
- storybook list 속성변경만 추가.
- list 자체를 추가하고싶었지만, 아직 문법을 이해하지 실패.

yg1110 committed:
Add ul component

- 채용 프로그램 ul 컴포넌트 추가.
- our-feature ul 컴포넌트 추가.
- storybook list 속성변경만 추가.
- list 자체를 추가하고싶었지만, 아직 문법을 이해하지 실패.

tkddn204 commented on src/Components/Ul/index.tsx:
```diff
@@ -0,0 +1,34 @@
+import React, { FC } from 'react'
+import styled from 'Application/Theme'
+
+import { Story } from '@storybook/react/types-6-0'
```
스토리는 스토리북 전용이기 때문에 `FC` 대신 사용해서는 안됩니다. 수정 부탁드립니다.

tkddn204 commented on src/Pages/Main/index.tsx:
```diff
@@ -1,6 +1,26 @@
+import { PeriodLl, TabItem } from 'Components/Li'
 
-const Main: FC = () => <Layout>Hello, Programmers-clone!</Layout>
+const Main: FC = () => (
+  <>
+    {/* <Layout>Hello, Programmers-clone!</Layout> */}
+    <PeriodUl>
+      <PeriodLl>접수 : 20년 08월 27일 14:00 - 11월 05일 18:00</PeriodLl>
+      <PeriodLl>대회 : 20년 09월 10일 19:30 - 11월 05일 22:30</PeriodLl>
+    </PeriodUl>
+
+    {/*
+            기존 코드에는 TabItem아래에 h4와 p태그가 자식컴포넌트로 존재합니다.
+            현재는 테스트를 하기위해 제목text만 넣어놨습니다.
+        */}
+    <ListFeatureUl>
+      <TabItem>코드에 집중하는 채용 프로그램</TabItem>
+      <TabItem>실전을 위한 트레이닝 시스템</TabItem>
+      <TabItem>코드리뷰 중심의 교육과정</TabItem>
+    </ListFeatureUl>
+  </>
+)
```
테스트하는 부분을 실제 프로그램에 적용하지 않고 일단 스토리북으로 내보내는 방법은 어떨까요?

tkddn204 reviewed (commented):
-

tkddn204 reviewed (changes requested):
한 컴포넌트에는 하나 이상의 스토리가 있는 것을 권장합니다. 다음 커밋에 Li 컴포넌트의 스토리북 파일을 만들어주실수 있을까요?

tkddn204 commented on src/Components/Li/index.tsx:
```diff
@@ -0,0 +1,35 @@
+  cursor: pointer;
+`
+
+// 채용프로그램의 리스트
+const PeriodLl: FC = (props) => {
+  const { children } = props
+  return <Li>{children}</Li>
+}
+
+// our-feature의 리스트
+const TabItem: FC = (props) => {
+  const { children } = props
+  return <TabStyledItem>{children}</TabStyledItem>
+}
```
이렇게 나누는 목적에 대해서 이야기를 하지 않았군요.
제가 이렇게 A 태그의 스타일을 받아서 나누는 목적은
용도(A에 사용, B에 사용, ...)가 아닌 스타일 변경(A 태그의 폰트 사이즈 변경, 폰트 색 변경, ...) 이었습니다.
나중에 컴포지션을 만들 때 따로따로 만들 생각이었는데, 이 부분도 나중에 얘기해봅시다.

HyeanWoo replied:
`PeriodLl`와 `TabItem`를 사용하지않고 불러오는 페이지에서 children을 넣어줘도 차이가 없어보입니다.

tkddn204 commented on src/Components/Ul/index.tsx:
```diff
@@ -0,0 +1,34 @@
+  color: #5f7f90;
+`
+
+// 채용프로그램의 리스트
+const PeriodUl: FC = (props) => {
+  const { children } = props
+  return <PeriodStyledUl>{children}</PeriodStyledUl>
+}
+
+// our-feature의 리스트
+const ListFeatureUl: Story = ({ children }) => {
+  return <Ul>{children}</Ul>
+}
```
이렇게 나누는 목적에 대해서 이야기를 하지 않았군요.
제가 이렇게 A 태그의 스타일을 받아서 나누는 목적은
용도(A에 사용, B에 사용, ...)가 아닌 스타일 변경(A 태그의 폰트 사이즈 변경, 폰트 색 변경, ...) 이었습니다.
나중에 컴포지션을 만들 때 따로따로 만들 생각이었는데, 이 부분도 나중에 얘기해봅시다.

HyeanWoo replied:
여기도 위에 질문과 같은 내용이네요

tkddn204 commented on src/Components/Ul/index.tsx:
```diff
@@ -0,0 +1,34 @@
+}
+
+const Ul = styled.ul<StyledProps>`
+  list-style: none;
+  list-style: ${(props) => props.listStyle};
```
아래 코드처럼 한 줄로 바꿔보는 건 어떨까요?
`list-style: ${(props) => props.listStyle && 'none'};`

yg1110 committed:
[#13]add ul component

Ul 스토리북 수정
	- text
	- list style
	- text에 따른 item 추가 및 수정
Li 스토리북 추가
	- text
	- color
main Page 테스트 코드 제거

tkddn204 commented:
좋습니다 👍

HyeanWoo commented on src/Components/Li/index.tsx:
```diff
@@ -1,35 +1,12 @@
 
-const Li = styled.li`
+export interface StyledProps {
+  color?: '5F7F90'
```
여기에는 타입적는거 같습니다

yg1110 replied:
아 다음 커밋에 바꿀게요.

HyeanWoo reviewed (approved):
확인했습니다 ~~
